src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -191,6 +191,11 @@ object VariantSampleMatrix {
 
       p match {
         case Some(partitioner) =>
+          if (rdd.partitions.length != partitioner.numPartitions)
```

Comment:
Why not put an error message on the assert in OrderedRDD?  Fatal should be reserved for user-interpretable error messages.